tests/context/conftest.py

```diff
@@ -242,3 +242,815 @@ def ce_details_dynamic_event_deleted(ce_details_dynamic_event):
     ce_details_dynamic_event_deleted = ce_details_dynamic_event.copy()
     del ce_details_dynamic_event_deleted["events"]["reminder2"]
     return ce_details_dynamic_event_deleted
+
+
+@pytest.fixture()
+def collection_exercises_with_details():
```

Comment:
This has a lot of stuff we don't need. Notice up the top there is a slim line ce_details. The idea being that is the starting point, each fixture below extents depending on what it needs to test. It doesn't need anything that it is not testing, it's not going to the frontend so dates, events, samples etc aren't needed for everything. So if it's build_reporting_units_context your testing, call your fixture appropriately extent the CE base and just make sure the values are populated to test, everything else could be None.

It will reduce the code massively and make it easier to maintain

Reply:
Updated and slimmed down